presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlock.java

```diff
@@ -83,7 +83,7 @@ public VariableWidthBlock(int positionCount, Slice slice, int[] offsets, Optiona
     }
 
     @Override
-    protected final int getPositionOffset(int position)
+    public final int getPositionOffset(int position)
```

Comment:
We are opening these methods here. It should be fine to do something similar for `ArrayType` / `MapType` / etc.:

Reply:
@arhimondr Andrii, the reason they are exposed here was because there was no ColumnarXXX for VariableWidthBlock, and I marked it as a TODO item to create a ColumnarSlice and expose these fields there. I've heard different opinions whether we can expose these fields like this and I think we need some better discussion on it.  cc @wenleix